src/components/submission-common.js

```diff
@@ -373,18 +373,34 @@ const validateRequiredHasSource = (inputSource, inputType) => {
   } else return { type: 'error', message: 'This attribute is required' }
 }
 
-const validateRecordLookup = (source, recordAttributes) => {
-  if (source) {
-    if (source.type === 'record_lookup' && !recordAttributes.includes(source.record_attribute)) {
-      return { type: 'error', message: 'This attribute doesn\'t exist in the data table' }
-    }
-    if (source.type === 'object_builder') {
-      if (source.fields) {
-        const fieldsValidated = _.map(field => field && validateRecordLookup(field.source, recordAttributes) === true, source.fields)
+export const typeMatch = (cbasType, wdsType) => {
```

Comment:
Does this need cases for type `File` and `JSON`? I think WDS does have `FILE` and `JSON` data types while CBAS has `File` and `Object` types as data types

Reply:
I will look into this, and it's worth noting that any cases not explicitly handled will pass the comparison (so not including cases will err on the side of allowing the input instead of rejecting the input). Allowing any type input for File would be consistent with our Type a Value validation for that type (`isPrimitiveTypeInputValid`).

Reply:
Any reason not to include the `Struct` type in the check?

Reply:
This check will be performed fieldwise on Structs that use the object builder and have record lookups for their fields. I wasn't sure if there's a way to easily see what fields a WDS `JSON` has and cross-check with the struct fields CBAS expects, but I suppose it's better to restrict Struct lookups to only JSON objects rather than anything (even if we can't check further)

Reply:
Yeah I think we for Struct inputs and `Fetch from Data table` as source, we can ensure that the attribute has `JSON` type and then let CBAS handle if the fields in the JSON object are actually what Struct input needs. What do you think?

Reply:
Addressed in latest commit.